setup.py

```diff
@@ -42,6 +42,11 @@
                       "scikit-learn>=0.19,<2.0",
                       "tqdm>=4.0,<5.0",
                       "scikit-optimize>=0.5,<2.0",
+                      "xgboost>=0.72.1,<1.0",
```

Comment:
Usually we guard against 2.0 instead of 1.0 in the case of 0 major version since 1.0 is not expected to break the API any differently than the next minor version (see https://semver.org/). Although it's very cosmetic here since xgboost seems like it'll stick to its 0 major version forever.